src/models/ui.ts

```diff
@@ -8,6 +9,8 @@ export const UIModel = types
     leftNavExpanded: false,
     myWorkExpanded: false,
     error: types.maybeNull(types.string),
+    activeSection: types.maybeNull(types.reference(SectionModel)),
```

Comment:
Is there a reason to prefer `maybeNull()` to `maybe()` here? Given that MST has moved to make `maybe()` the default and others in the TypeScript/JavaScript communities [recommend against the use of null](https://basarat.gitbooks.io/typescript/docs/javascript/null-undefined.html#final-thoughts) generally I'd be inclined to prefer `maybe()` and limit our use of null to cases where there is a semantic difference between null and undefined.

Also, on the Precipitating Change project we ended up moving away from MST references because they throw exceptions on lookup failures rather than just (in this case) allowing the reference to become null/undefined. What we ended up doing was using a string member and then performing the lookup in `afterCreate()`, which could make the reference undefined if the lookup failed. Given that the reference is nice but not required, failing the app seems extreme and adding exception-handling adds complexity.

Reply:
To me there is always difference in semantic meaning between `undefined` and `null` inside a type definition.  Undefineds are optional whereas nullables are required with `null` denoting "no current value".  Both `error` and `activeSection` are required attributes of the type.

I wasn't aware of the exceptions if the reference failed.  I think we should merge as is and then we can discuss both the `maybeNull` issue and the `reference` issue after the demo.

Reply:
`maybe()` and `maybeNull()` have the same semantics in that they are required members of the model, but optional in the sense that they have default values if they are not provided by the caller on creation -- the difference is just whether that default value is null or undefined. (Aside: I would have named MST's `types.optional()`, which underlies both `maybe()` and `maybeNull()`, something like `types.defaultable()`.)